service_mesh/v2x/ossm-extensions.adoc

```diff
@@ -0,0 +1,166 @@
+
+.Supported languages
+|===
+| Language | Maintainer | Repository
+
+| C++ 
+| proxy-wasm team (Istio Community) 
+| link:https://github.com/proxy-wasm/proxy-wasm-cpp-sdk[proxy-wasm/proxy-wasm-cpp-sdk]
+
+| Rust 
+| proxy-wasm team (Istio Community) 
+| link:https://github.com/proxy-wasm/proxy-wasm-rust-sdk[proxy-wasm/proxy-wasm-rust-sdk]
+
+| AssemblyScript 
+| solo.io 
+| link:https://github.com/solo-io/proxy-runtime[solo-io/proxy-runtime]
+
+| Go 
+| tetrate.io 
+| link:https://github.com/tetratelabs/proxy-wasm-go-sdk[tetratelabs/proxy-wasm-go-sdk]
+|===
+
+=== Container Format
+
+You must have a `.wasm` file containing the bytecode of your WebAssembly module, and a `manifest.yaml` file in the root of the container filesystem to make your container image a valid extension image.
+
+.manifest.yaml
+[source,yaml]
+----
+schemaVersion: 1
+  
+name: <your-extension>
+description: <description>
+version: 1.0.0
+phase: PreAuthZ
+priority: 100
+module: extension.wasm
+----
+
+.manifest.yml Field Reference
+|===
+| Field | Description
+
+|schemaVersion
+|Used for versioning of the manifest schema. Currently the only possible value is `1`.
+
+|name
+|The name of your extension. This field is just metadata and currently unused.
+
+|description
+|The description of your extension. This field is just metadata and currently unused.
+
+|version
+|The version of your extension. This field is just metadata and currently unused.
+
+|phase
+|The default execution phase of your Extension. This is a required field.
+
+|priority
+|The default priority of your Extension. This is a required field.
+
+|module
+|The relative path from the container filesystem's root to your WebAssembly module. This is a required field.
+|===
+
+=== Example
+
+For a complete example that was built using the Rust SDK, take a look at the link:https://github.com/maistra/header-append-filter[header-append-filter]. It is a very simple filter that appends a header, `custom-header`, to all responses, with the value depending on its configuration.
+
+=== Enabling WebAssembly Extension support
+
+Support for WebAssembly Extensions to {ProductName} is currently in Tech Preview, so it must be explicitly enabled for your `ServiceMeshControlPlane`. Set `spec.techPreview.wasmExtensions.enabled` in your SMCPv2 to `true`. Here's an example:
+
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: openid-connect
+  namespace: istio-system
+spec:
+  techPreview:
+    wasmExtensions:
+      enabled: true
+----
+
+=== Deploying Extensions
+
+{ProductName} Extensions can be enabled using the `ServiceMeshExtension` resource. The following snippet is an example resource.
+
+[source,yaml]
+----
+apiVersion: maistra.io/v1alpha1
+kind: ServiceMeshExtension
+metadata:
+  name: header-append
+  # Deploy to the Control Plane Namespace to enable globally, 
```

Comment:
Do we need these comments in the example if they're repeated in the table below?  I think the code comments could be merged into the table.